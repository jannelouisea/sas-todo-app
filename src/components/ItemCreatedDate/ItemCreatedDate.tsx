import Typography from '@mui/joy/Typography';
import HourglassBottomIcon from '@mui/icons-material/HourglassBottom';
import moment from 'moment';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import {
    MUIVariant,
    MUIText,
} from 'src/static/enums';

type Props = {
    createdAt: number,
    isCompleted: boolean,
    isOlder: boolean,
}

function ItemCreatedDate({ createdAt, isCompleted, isOlder }: Props) {

    const muiTextStyles = { padding: 0, paddingTop: '.25rem', marginLeft: '.25rem', marginRight: '.25rem' };
    const muiIconStyles = { fontSize: '1rem' };

    const isOlderIconClassNames = classNames({ 'hidden': isCompleted || (!isOlder) });

    return (
        <div className='item-created-date'>
            <Typography
                textColor={isCompleted ? 'neutral.700' : 'neutral.500'}
                level={MUIText.Label}
                variant={MUIVariant.Plain}
                sx={muiTextStyles}
            >
                {`Created ${moment.unix(createdAt).format('lll')}`}
            </Typography>
            <div className={isOlderIconClassNames}>
                <HourglassBottomIcon sx={muiIconStyles} />
            </div>
        </div>
    );
}

ItemCreatedDate.propTypes = {
    createdAt: PropTypes.number,
    isCompleted: PropTypes.bool,
    isOlder: PropTypes.bool,
}

ItemCreatedDate.defaultProps = {
    isCompleted: false,
    isOlder: false,
}


export default ItemCreatedDate;