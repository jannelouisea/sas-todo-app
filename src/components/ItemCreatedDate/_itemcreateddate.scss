.item-created-date {
    display: flex;
    align-items: center;
}