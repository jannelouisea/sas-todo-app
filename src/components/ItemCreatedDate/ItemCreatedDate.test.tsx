import ItemCreatedDate from './ItemCreatedDate';
import renderer from 'react-test-renderer';
import moment from 'moment';

describe('ItemCreatedDate', () => {
    const date = moment('2023-07-18').toDate();

    it('renders correctly', () => {
        const tree = renderer
            .create(<ItemCreatedDate createdAt={date} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders proper styles for completed item', () => {
        const tree = renderer
            .create(<ItemCreatedDate createdAt={date} isCompleted />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders proper styles for newer incomplete item', () => {
        const tree = renderer
            .create(<ItemCreatedDate createdAt={date} isOlder={false} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders properly for older incomplete item', () => {
        const tree = renderer
            .create(<ItemCreatedDate createdAt={date} isOlder />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});