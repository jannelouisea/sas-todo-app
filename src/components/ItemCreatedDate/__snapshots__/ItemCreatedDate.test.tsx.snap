// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCreatedDate renders correctly 1`] = `
<div
  className="item-created-date"
>
  <p
    className="MuiTypography-root MuiTypography-body3 MuiTypography-colorNeutral MuiTypography-variantPlain css-gfqtt1-JoyTypography-root"
  >
    Created Jul 18, 2023 12:00 AM
  </p>
  <div
    className="hidden"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17e890r-MuiSvgIcon-root"
      data-testid="HourglassBottomIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m18 22-.01-6L14 12l3.99-4.01L18 2H6v6l4 4-4 3.99V22h12zM8 7.5V4h8v3.5l-4 4-4-4z"
      />
    </svg>
  </div>
</div>
`;

exports[`ItemCreatedDate renders proper styles for completed item 1`] = `
<div
  className="item-created-date"
>
  <p
    className="MuiTypography-root MuiTypography-body3 MuiTypography-colorNeutral MuiTypography-variantPlain css-1oc79or-JoyTypography-root"
  >
    Created Jul 18, 2023 12:00 AM
  </p>
  <div
    className="hidden"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17e890r-MuiSvgIcon-root"
      data-testid="HourglassBottomIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m18 22-.01-6L14 12l3.99-4.01L18 2H6v6l4 4-4 3.99V22h12zM8 7.5V4h8v3.5l-4 4-4-4z"
      />
    </svg>
  </div>
</div>
`;

exports[`ItemCreatedDate renders proper styles for newer incomplete item 1`] = `
<div
  className="item-created-date"
>
  <p
    className="MuiTypography-root MuiTypography-body3 MuiTypography-colorNeutral MuiTypography-variantPlain css-gfqtt1-JoyTypography-root"
  >
    Created Jul 18, 2023 12:00 AM
  </p>
  <div
    className="hidden"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17e890r-MuiSvgIcon-root"
      data-testid="HourglassBottomIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m18 22-.01-6L14 12l3.99-4.01L18 2H6v6l4 4-4 3.99V22h12zM8 7.5V4h8v3.5l-4 4-4-4z"
      />
    </svg>
  </div>
</div>
`;

exports[`ItemCreatedDate renders properly for older incomplete item 1`] = `
<div
  className="item-created-date"
>
  <p
    className="MuiTypography-root MuiTypography-body3 MuiTypography-colorNeutral MuiTypography-variantPlain css-gfqtt1-JoyTypography-root"
  >
    Created Jul 18, 2023 12:00 AM
  </p>
  <div
    className=""
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17e890r-MuiSvgIcon-root"
      data-testid="HourglassBottomIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m18 22-.01-6L14 12l3.99-4.01L18 2H6v6l4 4-4 3.99V22h12zM8 7.5V4h8v3.5l-4 4-4-4z"
      />
    </svg>
  </div>
</div>
`;
